kaizhilam/ProjectCreature
# Armbow fully integrated, weapon wielded will now mirror inventory

* Adding clear materials for knife and armbow (not final look)
* Added armbow prefab and script
* Armbow glows
* Players attack method will now entirely depend on which weapon is being wielded
* Pressing 1 will equip weapon in slot 1 (if any). Same for 2 and 3
* Armbow can now be wielded, attached to forearm bone
* Script on armbone added so it can be found easily and its clearer in code which bone is being searched for
* Swapping inv weapons will update player weapon is selected hotbar index weapon is changed
* If item added to slot where player is indexed at, it will wield that weapon
* Each SlottedItem can now provide a local position, local rotation, local scale and bone index to add the item to
* 0 bone index refers to hand, -1 refers to hands parent bone (the arm)
* InputManager now broadcasts when 1,2 or 3 is pressed passing in the number pressed (int) as parameter
* Inventory now holds the value of which index of the hotbar the player has selected
* Scripts put on weapon prefabs now require the prefab in the inspector (for instantiation)

--Known bugs/issues--
* StateMachine will cause bulk errors after a certain time
* No visualisation for which slot player is on in hotbar

--How to test this branch--
* Pick up armbow and knife
* Player should have slot1 item equipped
* if you swap slot1 for the other weapon, player should switch weapon on hand
* if you drag weapon off slot1, it should unequip the weapon
* Pressing 2 will equipped weapon in slot2
* Attacking with knife should be able to kill dino
* Attacking with armbow should print message to log
